### ISSUE TITLE
Feature: need a key shortcut for "square your sets"

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#495]
...maybe also one for "square your sets and HEADS START".
This is done SO often when I am testing out sequences, that it begs for a keyboard shortcut.

Something like OPTION-CMD-S, maybe?

It should also put the focus into the text entry box.

**Comment by mpogue2** [comment]
This key combination only has to be valid within the SD tab.  Probably should have an entry in the SD menu, too, which makes it discoverable.

**Comment by mpogue2** [comment]
Note that this is different from #482.  That one added an entry to the context dropdown menu (right click on the sequence), and no shortcut key.  I'd really like to have a single key shortcut that squares the set, and it should be listed in the SD menu.

**Comment by danlyke** [comment]
In a127d64de94d9fc97242b2f908df19b0431336b6

Added configurable key sequences for abort this sequence, heads start,
heads 1p2p, and heads start+square thru 4.

In 55c1011526f4a4e965173f2f20ee4319ff76c7ec

created an SD->Shortcuts submenu with accelerator key
bindings.


In 6047126d312ce5cf212a7ca784529862690be02d

added default key bindings.

**Comment by mpogue2** [comment]
Nice!
Couple of bugs to fix:
- I have the menu items now (cool!), but I do not see any accelerator keys in the menu.  I think it would be better to have these as part of the main SD menu, instead of being in a shortcuts submenu, but if this is difficult, it's OK the way it is.  But, without the accelerator keys showing, the interface is not discoverable, so I'd like to get those in there (like on other menu items).
- Corner box shortcut does not take me to corner box, it takes me to Square Your Set + Heads Start.  I should say, it took me to corner box once out of about 50 tries.  So, it's doing something, just not reliably getting me to corner box.

**Comment by mpogue2** [comment]
Oh, another question:  the hashes don't match anything I can see in gitkraken in the master branch in the repo.  Am I looking at this wrong?

The 4 changes you made look to me like:
f62e9e374adbd967555297a47d6d6cf547a8cca8
33bf158247f17db4748912789c81f5e939d9b868
a2d882b9ebb796d7482cba1c534811aa27f7a6f2
22b44106bc62bf055a460e62be96f1baa89394af

**Comment by danlyke** [comment]
Hmmm... Okay, then, 4 issues:

1. I don't know what's happening with the inconsistency on the corner
box. I suspect I've got some state transition in SD to reset the system
wrong. I need to be restarting the thread on program changes anyway,
'cause SD sets some things up differently on different programs, so
that's a slightly longer thing.

2. I think the hash changes came 'cause I did a `git pull --rebase`
rather than a merge. My bad.

3. Please feel free to move those suckers around and the menu
structure you want. I bow to your aesthetic sense.

4. I'll figure out why the accelerators aren't getting set.

**Comment by mpogue2** [comment]
786a76d10fd461d686c0c39c65a6ca5cd1a0ee29 fixes a couple of problems.
- #elsex was a compile bug
- The code for corner box LOOKED correct, but it still didn't work reliably (it did work sometimes!).  I'm not sure why that code didn't work, but I played around with it a bit, and this new code appears to be reliably getting us to a corner box now.  I tried rapidly using all 4 new shortcuts, and they worked every time.
- I added some code to explicitly clear the input text box, AND to put the focus there after the shortcut is executed.
- Verified that these new menu items are in the SD main menu and discoverable now (thanks!)
- Verified that accelerator shortcuts show up now (thanks!!)
- And, I learned something about git rebase, too!  :-)

SD is so much easier to use now, when I am running sequences through SD as fast as I can (which I do a lot).  This is even better than what I had imagined -- thanks, Dan!!

Closing the issue, because I think it's been delivered completely.  I'll open a separate issue for the SD "reset thread on program level change" (unless there is one open already).
